tikv/mvcc.go

```diff
@@ -58,20 +58,27 @@ func NewMVCCStore(db *badger.DB, dataDir string) *MVCCStore {
 	if err != nil {
 		log.Fatal(err)
 	}
-	store.wg.Add(3)
+	// run all the workers
 	go store.writeDBWorker.run()
 	go store.writeLockWorker.run()
-	rbGCWorker := rollbackGCWorker{store: store}
-	go rbGCWorker.run()
+	go func() {
+		rbGCWorker := rollbackGCWorker{store: store}
+		rbGCWorker.run()
+	}()
+
+	// mark worker count
+	store.wg.Add(3)
```

Comment:
It's safer to add WaitGroup count before the goroutines run.

Reply:
Why was that? we are still inside the current initialize function.

Reply:
In this case, it is safe, but in general coding style, we may run into wg count decrease to a negative value in some cases.

Reply:
We should never use WaitGroup until we initialize it successful. I prefer to keep this clear.